lib/ansiblelint/runner.py

```diff
@@ -101,9 +87,25 @@ def run(self) -> List:
                 "Examining %s of type %s",
                 ansiblelint.file_utils.normpath(file['path']),
                 file['type'])
-            matches.extend(self.rules.run(file, tags=set(self.tags),
-                           skip_list=self.skip_list))
+            matches = matches.union(
+                self.rules.run(file, tags=set(self.tags),
+                               skip_list=self.skip_list))
         # update list of checked files
         self.checked_files.update([x['path'] for x in files])
 
-        return sorted(set(matches))
+        return sorted(matches)
+
+    def _emit_matches(self, files: List) -> Generator[MatchError, None, None]:
```

Comment:
Oh! It looks like the semantical goal of this block is loading the playbooks. So it'd probably be better to say
```suggestion
    def _emit_playbook_load_errors(self, files: List) -> Generator[MatchError, None, None]:
```

Reply:
It would be incorrect load errors is just some of the possible MatchErrors, it would be misleading.

Reply:
FWIW it is misleading now.